### PR TITLE
Test against multiple specific clang/gcc versions

### LINKED ISSUE #208: Pick a few compiler versions and test against them
Pick a few "supported" compiler versions and test against them on Travis. Right now we are just testing against what is there, whatever that is.

This does not mean that older or other compilers won't work, since tommath is very conservative in that regard. They are just not tested during CI runs.

### PR DESCRIPTION
**Pull request by minad**
Fixes #208

### CODE REVIEW

**Comment by minad** [comment]
@sjaeckel I added the Wconversion warnings here - enabled only on recent clang-7 which does not give unneessary false positives.

File: .travis.yml [hunk 1]
```diff
@@ -31,15 +31,15 @@ branches:
...
-# Additional installs are Valgrind for the memory-tests
-# and gcc-multilib for the compilation of the different
-# architectures.
+# Additional installs: Valgrind for memory tests.
 install:
   - sudo apt-get update -qq
   - sudo apt-get install valgrind
-  - sudo apt-get install gcc-multilib
+  - apt-cache search gcc | grep '^gcc-[0-9\.]* '
+  - apt-cache search clang | grep compiler
```
Comment (czurnieden) [inline .travis.yml]: I would install them individually instead. The newer ones won't be a problem but the older ones, like e.g.: gcc-4.7 might. Or set the distribution which would be the safest variant, I think.

**Comment by minad** [comment]
> More castings than in a trauma centre in Aspen/Colorado in high season but otherwise&hellip

But only in the tests suite :)

File: .travis.yml (hunk 1 above)
Comment (minad) [reply .travis.yml]: The compileds are installed separately. The distro is fixed. These lines just print the available compilers in the package database.

**Pull request merged**